### PR TITLE
perf(array-repeat-strategy): call array.slice with all arguments

### PR DESCRIPTION
**Pull request by Gheoan**
Also, is `array.slice` the fastest way to clone an array?

### CODE REVIEW

**Comment by EisenbergEffect** [comment]
@martingust @jdanyow What do you guys think?

File: src/array-repeat-strategy.js [hunk 1]
```diff
@@ -109,11 +109,11 @@ export class ArrayRepeatStrategy {
         let {index, removed, addedCount} = splices[i];
         mergeSplice(repeat.__queuedSplices, index, removed, addedCount);
       }
-      repeat.__array = array.slice(0);
+      repeat.__array = array.slice(0, array.length);
```
Comment (alvarezmario) [inline src/array-repeat-strategy.js]: @Gheoan it seems is the best option, and also, 0 and array.lenght are the default values for slice, so you can safely remove them

Comment (alvarezmario) [inline src/array-repeat-strategy.js]: Well, in fact, according to this http://jsperf.com/new-array-vs-splice-vs-slice/113, `slice` is not the fastest, is at the medium of the table. Using a while loop seems to be faster, but is really worth?

**Comment by Gheoan** [comment]
Looking at the [spec](http://www.ecma-international.org/ecma-262/6.0/#sec-array.prototype.slice) again, I was wrong :sweat_smile:, passing the 2nd argument causes an extra call to `ToInteger` so omitting it should be faster. 
> If _end_ is **undefined**, let _relativeEnd_ be _len_; else let _relativeEnd_ be ToInteger(end).

**Comment by EisenbergEffect** [comment]
It's probably very browser specific. I just tested FF on Mac and slice was faster than the while loop, but only by a little bit.

**Comment by EisenbergEffect** [comment]
I think we can leave it as is for now.

**Pull request closed without merging**